frontend/catalyst/api_extensions/quantum_operators.py

```diff
@@ -723,3 +723,11 @@ def _check_no_measurements(tape: QuantumTape) -> None:
         else:
             if isinstance(op, MidCircuitMeasure):
                 raise ValueError(msg)
+
+
+## API ##
+def StatePrep(state, wires, id=None)-> DynamicJaxprTracer:
+    """Catalyst version of PennyLane"""
+    
+    raise TypeError("StatePrep is not implemented yet")
```

Comment:
Could it be possible to not implement a new operator? Most other quantum operators are lowered to the CustomOp in JAXPR without needing a dedicated API that matches PennyLane's. To do this, I would imagine modifying the toml file and then lowering to a JAXPR primitive. I think you mentioned trying this, can you elaborate a bit?

Reply:
This operator does not fit the `CustomOp` mold, because it accepts a complex vector as input (rather than our usual sequence of float parameters).

Reply:
Yes we definitely shouldn't add our own function for doing `StatePrep`, the existing operator simply needs to be supported by the quantum function lowering we do in `jax_tracer.py`

Reply:
Have a look at the `bind_native_operation` function to add a case for this operator there.

Reply:
Makes sense. The new operator was discarded. I will lower it to a JAX primitive now. In progress.